routes/dao_coin_exchange.go

```diff
@@ -346,6 +346,56 @@ func buildDAOCoinLimitOrderResponse(
 // Helper functions to calculate price and exchange rates for DAO coin limit orders
 ///////////////////////////////////////////////////////////////////////////////////
 
+// GetBestAvailableExchangeRateCoinsToBuyPerCoinToSell computes the best available decimal string exchange rate at which
+// the market is able to exchange one base unit of the selling coin pair for the buying coin. Since we are interested
+// in computing the best exchange rate for the selling coin, the denominator for the output will always be the selling coin.
+//   Example: given buying coin B, and selling coin S, an output exchange rate of "1.5" implies an exchange rate of
+//            (1.5 coin B) per (1 coin S).
+// This function can support any arbitrary coin pair, but is most useful for markets where one coin is always considered
+// the denominating coin (ex: DAO coin <> DESO). In such cases, this computes the best available ask price.
+func GetBestAvailableExchangeRateCoinsToBuyPerCoinToSell(
+	fes *APIServer,
```

Comment:
```suggestion
func (fes *APIServer) GetBestAvailableExchangeRateCoinsToBuyPerCoinToSell(
```
We typically make these functions on the APIServer struct instead of pass the APIServer as a param